src/main/java/com/exadel/discountwebapp/history/mapper/HistoryMapper.java

```diff
@@ -0,0 +1,36 @@
+package com.exadel.discountwebapp.history.mapper;
+
+import com.exadel.discountwebapp.history.vo.HistoryRequestVO;
+import com.exadel.discountwebapp.history.vo.HistoryResponseVO;
+import org.javers.core.Changes;
+import org.javers.core.diff.changetype.ValueChange;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Component
+public class HistoryMapper {
+    private final HistoryResponseVO historyResponseVO = new HistoryResponseVO();
+
+    public List<HistoryResponseVO> toVO(Changes changes, HistoryRequestVO historyRequestVO) {
+
+        ArrayList<HistoryResponseVO> build = new ArrayList<>();
+
+        for (var i = 0; i < changes.size(); i++) {
```

Comment:
if the list "changes" contains changes with type other then "ValueChange" then next code will get IndexOutOfBoundsException:
changes.getChangesByType(ValueChange.class).get(i)

because total size of "changes" will be bigger then "changes.getChangesByType(ValueChange.class)"

we need to iterate over changes.getChangesByType(ValueChange.class)

Reply:
changed